setup/building-es2k-stratum-deps.md

```diff
@@ -0,0 +1,221 @@
+export SDKTARGETSYSROOT=$SYSROOT
+export PKG_CONFIG_SYSROOT_DIR=$SYSROOT
+export PKG_CONFIG_PATH=$SYSROOT/usr/lib64/pkgconfig:$SYSROOT/usr/lib/pkgconfig:$SYSROOT/usr/share/pkgconfig
+[ -z "$ES2K_SAVE_PATH" ] && export ES2K_SAVE_PATH=$PATH
+export PATH=$AARCH64/bin:$ES2K_SAVE_PATH
+```
+
+In the listing above, you will need to provide values for these variables:
+
+- `ACC_SDK` - install path of the ACC-RL SDK (for example,
+  `$HOME/p4cp-dev/acc_sdk`)
+- `P4CPBASE` - path to the local networking-recipe directory (for example,
+  `$HOME/p4cp-dev/ipdk.recipe`)
+
+From these paths, the setup script derives:
+
+- `AARCH64` - path to the directory containing the AArch64
+  cross-compiler suite
+- `SYSROOT` - path to the sysroot directory, which contains AArch64
+  header files and binaries
+
+These directories are part of the ACC SDK.
+
+The setup script exports the following variables, which are used by CMake
+and the helper script:
+
+- `SDKTARGETSYSROOT` - path to the sysroot directory
+- `CMAKE_TOOLCHAIN_FILE` - path to the CMake toolchain file
+- `PKG_CONFIG_PATH` - search path for `pkg-config` to use when looking for
+  packages on the target system
+- `PKG_CONFIG_SYSROOT_DIR` - path to the sysroot directory, for use by
+  `pkg-config`
+
+The setup script also adds the directory containing the cross-compiler
+executables to the system `PATH`.
+
+> **Note:** The ACC-RL SDK includes its own setup file
+> (`environment-setup-aarch64-intel-linux-gnu`). We strongly recommend
+> that you *not* use this file when building the Stratum dependencies
+> or P4 Control Plane.
+>
+> The SDK setup file is intended for use with GNU Autotools. Some of
+> the environment variables it defines affect the behavior of the C and
+> C++ compilers and the linker. These definitions may interfere with the
+> CMake build in non-obvious ways.
+
+## 6. Building the Target Dependencies
+
+You will need to pick an install location for the target dependencies.
+This will typically be under the sysroot directory structure. For
+example, the `opt` subdirectory will become the root-level `/opt`
+directory when the file structure is copied to the E2100 file system.
+
+The `setup` directory includes a helper script (`make-cross-deps.sh`) that
+can be used to build the Target dependencies.
+
+- The `--help` (`-h`) option lists the parameters the helper script supports
+
+- The `--dry-run` (`-n`) option displays the parameter values without
+  running CMake
+
+You will need to provide the helper script with the path to the Host
+dependencies (`--host`) as well as the install prefix (`--prefix`).
+
+### Target build
+
+Source the file that the defines the [target build environment variables](#5-defining-the-target-build-environment).
+
+```bash
+source es2k-setup.env
+```
+
+Remove the `build` directory from the previous build.
+
+```bash
+rm -fr build
+```
+
+Now run the build script:
+
+```bash
+./make-cross-deps.sh --host=HOSTDEPS --prefix=PREFIX
```

Comment:
Maybe, i am missing some context here.
"make-cross-deps.sh" script only need '--prefix' and '--toolchain' parameters right?

Also, will the grpc patch would have applied while building HOST, will it also work fine when we run this script?

Reply:
`make-cross-deps.sh` requires host dependencies (we use the native protobuf compiler to compile .proto to .cc and .h), install prefix, and toolchain (which is normally provided by the CMAKE_TOOLCHAIN_FILE environment variable). `--toolchain` exists primarily to allow you to override the environment variable.

I've been having problems with the --force parameter since we updated grpc. When I was testing out command sequences, I discovered that I could delete the `build` directory and do another build, and it skipped the download and patch steps, so that's what I described in the document.

Reply:
This behavior appears to be related to specifying a full Git SHA instead of a branch or tag when you download.

Reply:
I just tried this again. It's re-downloading grpc (because one of the files has been patched and therefore doesn't match an expected signature?), but it apparently doesn't perform the patch step. But that's okay, because the patch only applies to the host build (something that hadn't occurred to me before)!

Reply:
I've created a Jira task to address this issue: https://jira.devtools.intel.com/browse/NVFMSA-8984